nemo/collections/multimodal/speech_llm/data/lhotse_dataset.py

```diff
@@ -0,0 +1,517 @@
+            if is_canary_tokens_augment:
+                if simple_augment:
+                    template = 'Transcribe the spoken content to written <|SLANG|> text, then translate this to <|TLANG|> text, <|PNC|>'
+                else:
+                    template = 'Transcribe the spoken content to written <|SLANG|> text, then translate this to English text, then translate this to <|TLANG|> text, <|PNC|>'
+            else:
+                template = 'Translate the spoken <|SLANG|> content to written <|TLANG|> text, <|PNC|>'
+        else:
+            assert False, 'Unknown task {}'.format(prompt)
+        return template
+
+    def get_pnc(text):
+        if text == "<|nopnc|":
+            pnc = 'ignoring punctuations and capitalization'
+        elif text == "<|pnc|":
+            pnc = 'with punctuations and capitalizations'
+        else:
+            assert False, 'Unknown pnc {}'.format(prompt)
+        return pnc
+
+    if len(ps) == 6:
+        source_lang = get_lang(ps[1])
+        target_lang = get_lang(ps[3])
+        pnc = get_pnc(ps[4])
+        task = get_task_template(ps[2], is_canary_tokens_augment)
+        task = task.replace('<|SLANG|>', source_lang)
+        task = task.replace('<|TLANG|>', target_lang)
+        task = task.replace('<|PNC|>', pnc)
+    else:
+        task = 'Above is not speech.'
+    return task
+
+
+class LhotseAudioQuestionAnswerDataset(torch.utils.data.Dataset):
+    """
+    This dataset is based on Lhotse ASR dataset from ``audio_to_text_lhotse.py``
+    and ``TarredAudioQuestionAnswerDataset`` from ``audio_text_qa_dataset.py``.
+
+    Unlike native NeMo datasets, Lhotse dataset defines only the mapping from
+    a CutSet (meta-data) to a mini-batch with PyTorch tensors.
+    Specifically, it performs tokenization, I/O, augmentation, and feature extraction (if any).
+    Managing data, sampling, de-duplication across workers/nodes etc. is all handled
+    by Lhotse samplers instead.
+
+    Args:
+        text_processor: TextProcessing object
+        default_question: Default question to use if no question is provided
+        tokens_to_generate: Number of tokens to generate during inference
+        pad_to_max_length: Whether to pad the input to the max sequence length. If False, will pad to the max length of the current batch.
+        max_seq_length: Maximum sequence length for each dataset examples. Examples will either be truncated to fit this length or dropped if they cannot be truncated.
+        canary_processor: Optional Canary processor to convert canary tokens to text
+        convert_canary_prompt_to_text: Whether to convert canary prompt to text
+        prepend_to_exist_question: Optional string to prepend to existing question
+        canary_tokens_augment_ratio: Ratio of canary tokens to augment
+        random_context_prob: Probability of injecting random context
+        random_context_positive_percent: Percentage of positive random context
+    """
```

Comment:
1. please make it more general, instead of canary-specific
2. please remove random context fields

Reply:
This class is more like a `LhotseCanaryAudioQuestionAnswerDataset` as it depends on a lot of Canary-specific stuffs.....I think we should make the class more general